src/resources/pet_api.py

```diff
@@ -0,0 +1,76 @@
+from requests import Response
+
+from src.data.pet.pet import Pet
+from src.json_schemas.pet_schemas import pet_schema
+from utils.base_api import BaseApi
+from utils.helpers import get_random_number
+
+
+class PetApi(BaseApi):
+    def __init__(self):
+        super().__init__()
+        self.__endpoint = '/pet'
+        self.__min_valid_pet_id = 1
+        self.__max_valid_pet_id = 999999999999999999
+
+    def get_pet(self, pet_id: int) -> Response:
+        """
+        Retrieves a pet by its ID.
+        :param pet_id: Pet ID
+        :return: API response
+        """
+        return self._get(endpoint=f'{self.__endpoint}/{pet_id}')
+
+    def add_pet(self, pet: Pet, is_unsupported_content_type: bool = False) -> Response:
+        """
+        Adds a new pet.
+        :param pet: Pet object to add
+        :param is_unsupported_content_type: Flag for unsupported content type
+        :return: API response
+        """
+        request_body = pet.to_dict()
+        headers = {'Content-Type': 'text/html'} if is_unsupported_content_type else {}
+        return self._post(endpoint=self.__endpoint, headers=headers, json=request_body)
+
+    def update_pet(self, pet: Pet | dict, is_unsupported_content_type: bool = False) -> Response:
+        """
+        Updates an existing pet.
+        :param pet: Pet object or dictionary to update
+        :param is_unsupported_content_type: Flag for unsupported content type
+        :return: API response
+        """
+        request_body = pet.to_dict()
+        headers = {'Content-Type': 'text/html'} if is_unsupported_content_type else {}
+        return self._put(endpoint=self.__endpoint, headers=headers, json=request_body)
+
+    def delete_pet(self, pet_id: int) -> Response:
+        """
+        Deletes a pet by its ID.
+        :param pet_id: Pet ID to delete
+        :return: API response
+        """
+        return self._delete(endpoint=f'{self.__endpoint}/{pet_id}')
+
+    def validate_pet_schema(self, response_body: dict) -> bool:
+        """
+        Validates the response schema for a pet.
+        :param response_body: Response body to validate
+        :return: True if valid, False otherwise
+        """
+        return self._validate_schema_of_response_body(response_body, pet_schema)
+
+    def generate_random_pet_id(self) -> int:
```

Comment:
I do not think this function is a part of PetApi. It is just a generator of a number, so please think about where to put it. It can be a separate Generators module